stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/OnlyMediaAttachmentsViewHolder.kt

```diff
@@ -3,52 +3,61 @@ package io.getstream.chat.android.ui.messages.adapter.viewholder
+import io.getstream.chat.android.ui.messages.adapter.view.AttachmentClickListener
+import io.getstream.chat.android.ui.messages.adapter.view.AttachmentLongClickListener
 import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.Decorator
 
 public class OnlyMediaAttachmentsViewHolder(
     parent: ViewGroup,
     decorators: List<Decorator>,
-    private val listenerContainer: MessageListListenerContainer?,
-    internal val binding: StreamUiItemMessageMediaAttachmentsBinding = StreamUiItemMessageMediaAttachmentsBinding.inflate(
-        parent.inflater,
-        parent,
-        false
-    )
+    listenerContainer: MessageListListenerContainer?,
+    internal val binding: StreamUiItemMessageMediaAttachmentsBinding =
+        StreamUiItemMessageMediaAttachmentsBinding.inflate(
+            parent.inflater,
+            parent,
+            false
+        )
 ) : BaseMessageItemViewHolder<MessageListItem.MessageItem>(binding.root, decorators) {
 
-    override fun bindData(data: MessageListItem.MessageItem, diff: MessageListItemPayloadDiff?) {
+    init {
         listenerContainer?.let { listeners ->
             binding.run {
                 root.setOnClickListener {
                     listeners.messageClickListener.onMessageClick(data.message)
                 }
-                root.setOnLongClickListener {
-                    listeners.messageLongClickListener.onMessageLongClick(data.message)
-                    true
+                mediaAttachmentsGroupView.attachmentClickListener = AttachmentClickListener {
+                    listeners.attachmentClickListener.onAttachmentClick(data.message, it)
                 }
-                mediaAttachmentsGroupView.listener =
-                    { attachment -> listeners.attachmentClickListener.onAttachmentClick(data.message, attachment) }
                 reactionsView.setReactionClickListener {
                     listeners.reactionViewClickListener.onReactionViewClick(data.message)
                 }
                 threadRepliesFootnote.root.setOnClickListener {
                     listeners.threadClickListener.onThreadClick(data.message)
                 }
+
+                root.setOnLongClickListener {
+                    listeners.messageLongClickListener.onMessageLongClick(data.message)
+                    true
+                }
+                mediaAttachmentsGroupView.attachmentLongClickListener = AttachmentLongClickListener {
+                    listeners.messageLongClickListener.onMessageLongClick(data.message)
+                }
             }
         }
+    }
 
+    override fun bindData(data: MessageListItem.MessageItem, diff: MessageListItemPayloadDiff?) {
         if (data.message.attachments.isMedia()) {
-            showAttachments(data.message.attachments)
+            binding.mediaAttachmentsGroupView.showAttachments(
+                *data.message
+                    .attachments
+                    .toTypedArray()
+            )
```

Comment:
We really ought to change this `showAttachments` method's parameter, makes no sense to have it be a `vararg` instead of a `List`